src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -227,7 +179,7 @@ EmberAfStatus OTARequestor::HandleAnnounceOTAProvider(
         return EMBER_ZCL_STATUS_FAILURE;
     }
 
-    ConnectToProvider(kQueryImage);
+    chip::DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(msToStart), StartDelayTimerHandler, this);
```

Comment:
Now I understand `StartDelayTimerHandler` (see my comment above).

Strongly suggest making all timer starting being a single method, rather than directly calling system layer from business logic in several places.

Reply:
Will be accounted for in Damian's change.